criu/pagemap.c

```diff
@@ -656,15 +656,23 @@ static int try_open_parent(int dfd, unsigned long id, struct page_read *pr, int
 {
 	int pfd, ret;
 	struct page_read *parent = NULL;
+	struct stat st;
 
 	/* Image streaming lacks support for incremental images */
 	if (opts.stream)
 		goto out;
 
-	pfd = openat(dfd, CR_PARENT_LINK, O_RDONLY);
-	if (pfd < 0 && errno == ENOENT)
+	/* Check if the parent symlink exists */
+	ret = fstatat(dfd, CR_PARENT_LINK, &st, AT_SYMLINK_NOFOLLOW);
+	if (ret < 0 && errno == ENOENT)
 		goto out;
 
+	pfd = openat(dfd, CR_PARENT_LINK, O_RDONLY);
+	if (pfd < 0 && errno == ENOENT) {
+		pr_perror("Can't open parent path\n");
+		goto err;
+	}
+
```

Comment:
These checks are repeated several times, maybe it's worth splitting them into a helper function?